Marlin/src/inc/SanityCheck.h

```diff
@@ -253,17 +253,18 @@
  */
 #ifndef USBCON
   #if ENABLED(SERIAL_XON_XOFF) && RX_BUFFER_SIZE < 1024
-    #error "XON/XOFF requires RX_BUFFER_SIZE >= 1024 for reliable transfers without drops."
+    #error "SERIAL_XON_XOFF requires RX_BUFFER_SIZE >= 1024 for reliable transfers without drops."
   #endif
 
   #if !IS_POWER_OF_2(RX_BUFFER_SIZE) || RX_BUFFER_SIZE < 2
     #error "RX_BUFFER_SIZE must be a power of 2 greater than 1."
   #endif
 
-  // 256 is the max limit due to uint8_t head and tail. Use only powers of 2. (...,16,32,64,128,256)
   #if TX_BUFFER_SIZE && (TX_BUFFER_SIZE < 2 || TX_BUFFER_SIZE > 256 || !IS_POWER_OF_2(TX_BUFFER_SIZE))
     #error "TX_BUFFER_SIZE must be 0, a power of 2 greater than 1, and no greater than 256."
   #endif
+#elif ENABLED(SERIAL_XON_XOFF)
+  #error "SERIAL_XON_XOFF is not supported on USB-native AVR devices."
```

Comment:
This will never get triggered due to the `#undef`. Either the code should have the `#undef` with a warning, or just this `#error`.

Reply:
I believe it was the other way around, the `#undef SERIAL_XON_XOFF` didn't get executed because compile didn't complete when SanityCheck threw the error.  I left the redundant check in Conditional_adv.h since it didn't seem to hurt, and might prevent future issue if something ever caused the SanityCheck to break.  But thinking about your comment, perhaps it is "smarter" to not try and be so smart, lest be met with unintended consequences (if compilation priority changes for some reason, for example).  I'll double check anyway to make sure current behavior throws the error instead of silent undef.

Reply:
Weird. The sanity check always comes after `Conditionals_post.h`.